test/e2e/certificates_test.go

```diff
@@ -0,0 +1,207 @@
+// The initial Kubelet CA certificate is valid for 1 year, from the date of the cluster installation.
+// Usually, the first rotation of the kubelet CA certificate is generated by the [API Server Operator](https://github.com/openshift/cluster-kube-apiserver-operator)
+// at 80% for the CA certificate lifecycle (1 year), which is 292 days, and removes the previous CA certificate after 365 days.
+func testCertificates(t *testing.T) {
+	// init test context
+	tc, err := NewTestContext()
+	require.NoError(t, err)
+
+	// TODO: Validate no workload disruption during certificate rotation/deletion.
+	// Deploy a Windows workload with WebServer, a service endpoint, and a job
+	// that continuously curls the service.
+	// After below tests pass, check for errors in the job.
+
+	t.Run("Kubelet CA rotation", tc.testKubeletCARotation)
+	t.Run("Kubelet CA deletion", tc.testKubeletCARemoval)
+}
+
+// testKubeletCARotation test the rotation of the Kubelet CA certificate. This test accounts for the certificate
+// rotation that takes place at 80% for the CA certificate lifecycle.
+// Note that the old certificate is still present in the cluster until day 365, so clients may be able to use it.
+func (tc *testContext) testKubeletCARotation(t *testing.T) {
+	// node logs must be available before rotation
+	err := checkJournalLogInWindowsNodes()
+	require.NoError(t, err, "error checking journal logs in Windows nodes")
+
+	// force CA rotation
+	err = tc.forceKubeApiServerCertificateRotation()
+	require.NoError(t, err)
+
+	// wait 5 seconds for the new certificate and add 3 seconds per node to allow
+	// WMCO to update the kubelet-ca.crt file on each node
+	nodeCount := len(gc.allNodes())
+	time.Sleep(time.Duration(5+3*nodeCount) * time.Second)
+
+	// node logs must be available after rotation
+	err = checkJournalLogInWindowsNodes()
+	assert.NoError(t, err, "error checking journal logs in Windows nodes after kubelet CA rotation")
+}
+
+// testKubeletCARemoval test the removal of the Kubelet CA certificate, it takes place after 365 days from cluster's
+// installation date.
+func (tc *testContext) testKubeletCARemoval(t *testing.T) {
+	// check node logs
+	err := checkJournalLogInWindowsNodes()
+	require.NoError(t, err, "error checking journal logs in Windows nodes")
+
+	// find initial Kubelet CA certificate
+	initialCAConfigMap, err := certificates.GetInitialCAConfigMap(context.TODO(), tc.client.Cache)
+	require.NoError(t, err)
+	initialCABytes, err := certificates.GetCAsFromConfigMap(initialCAConfigMap, certificates.CABundleKey)
+	require.NoError(t, err)
+	initialKubeApiServerSignerCert := findCertInCABundle(initialCABytes, "kube-apiserver-to-kubelet-signer")
+
+	// remove initial Kubelet CA certificate from kube-apiserver-to-kubelet-client-ca ConfigMap
+	err = tc.removeCertFromKubeAPIServerCA(initialKubeApiServerSignerCert)
+	require.NoError(t, err)
+
+	// wait 5 seconds for the Kube API server to process the certificate removal
+	time.Sleep(5 * time.Second)
```

Comment:
Will this introduce flakiness? Is there a way to `waitFor()` this to happen instead?

Reply:
Yes, we could track all the chain of events from kube-api-operator to MCO. However, it will add more complexity to the e2e test and a few extra API calls.

I ran several tests locally, and worst case it takes 5 seconds to apply the changes in vSphere and AWS (I did not test Azure) so I feel it's safer and simpler to just 🛌🏼 .

Reply:
I suggest making it slightly longer (seconds seems like a short window though that is just based on my speculation). Consider using one of our [retry constants](https://github.com/openshift/windows-machine-config-operator/blob/master/pkg/retry/retry.go#L9)